dali/pipeline/executor/executor.h

```diff
@@ -231,6 +264,155 @@ class DLL_PUBLIC Executor {
   using Executor::event_pool_;                             \
   using Executor::thread_pool_
 
+
+template <DALIOpType op_type>
+workspace_t<op_type>& Executor::get_workspace(workspace_owner_t& wo, OpPartitionId partition_idx) {
+  auto& ws_vec = std::get<static_cast<size_t>(op_type)>(wo);
+  return ws_vec[partition_idx];
+}
+
+
+template <DALIOpType op_type>
+workspace_t<op_type>& Executor::get_workspace(workspace_owner_t& wo, const OpNode &node) {
+  DALI_ENFORCE(node.op_type == op_type,
+               "Wrong variant of method selected. DALIOpType does not match.");
+  auto& ws_vec = std::get<static_cast<size_t>(op_type)>(wo);
+  return ws_vec[node.partition_index];
+}
+
+// We instantiate the operation of adding the input only for parent op_type and device
+// that are specifically allowed
+template <DALIOpType op_type, DALIOpType producer_type, DALITensorDevice device>
+en_if_t<allows_op_input<op_type>(producer_type) && allows_tensor_input<op_type>(device)> add_input(
```

Comment:
I suppose that `en_if_t` means `enable_if_t`. I'd rather splurge on having the extra characters :)

Reply:
...or make it `if_t` ;)

Reply:
Done